db/warehouse/migrate/20250331175933_create_cas_analytics_intermediate_tables.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+class CreateCasAnalyticsIntermediateTables < ActiveRecord::Migration[7.0]
+  def change
+    # Representation of sub-programs
+    create_table :cas_analytics_opportunity_categories do |t|
+      t.string :full_name, comment: 'concatenation of program name and sub-program name'
+      t.string :name
+      t.string :sub_project_name
+      t.string :program_type
+      t.references :subgrantee, index: false
+      t.string :subgrantee_name
+      t.references :sub_contractor, index: false
+      t.string :sub_contractor_name
+      t.references :hsa, index: false
+      t.string :hsa_name
+
+      t.timestamps
+    end
+
+    create_table :cas_analytics_opportunities do |t|
+      t.references :opportunity_category, index: false
+      t.references :unit
+      t.string :unit_name
+
+      t.timestamps
+    end
+
+    create_table :cas_analytics_clients do |t|
+      t.references :client, index: false
+      t.date :calculated_first_homeless_night
+      t.date :calculated_last_homeless_night
+
+      t.timestamps
+    end
+
+    create_table :cas_analytics_workflow_contacts do |t|
+      t.string :email
+      t.references :workflow, index: false
+      t.references :contact, index: false
+      t.string :contact_type
+
+      t.timestamps
+    end
+
+    create_table :cas_analytics_cas_users do |t|
+      t.string :email
+      t.references :agency, index: false
+      t.string :agency_name
+
+      t.timestamps
+    end
+
+    # Representation of client-opportunity-match combined with route
+    create_table :cas_analytics_workflows do |t|
```

Comment:
In the CE-in-HMIS world, this would be called a "referral". And a "workflow template" is equivalent to "match".
@eanders you mentioned that you would prefer if the same superset dataset was used for both CAS and CE-in-HMIS, perhaps backed by different (or the same?) analytics views. Can we do a quick huddle on this? I don't feel sure what direction we're going with CE data in superset, so I'm not sure how to review this.

Reply:
+1, that was my understanding as well.

Perhaps our plan is harmonize CE and CAS analytics views once CE is built? I imagine that building views from the ground up that support both is less likely to incur tech debt

Reply:
Yeah, let's huddle on this.  I thought a workflow template was a route and a workflow instance was a match.  I chose to ignore the template for now given that routes aren't currently in the DB and we should be able to generate the parts we need from the matches...  I very much might have these wrong, and it may also be that we want a slightly different shape for analysis than we do for end-user interactions.